comid/instance_test.go

```diff
@@ -67,3 +68,125 @@ func Test_RegisterInstanceType(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, instance.Bytes(), out2.Bytes())
 }
+
+func Test_NewBytesInstance_OK(t *testing.T) {
+	var testBytes = []byte{0x01, 0x02, 0x03, 0x04}
+
+	for _, v := range []any{
+		testBytes,
+		&testBytes,
+		string(testBytes),
+	} {
+		instance, err := NewBytesInstance(v)
+		require.NoError(t, err)
+		got := instance.Bytes()
+		assert.Equal(t, testBytes[:], got)
+	}
+}
+
+func Test_NewBytesInstance_NOK(t *testing.T) {
+	for _, tv := range []struct {
+		Name  string
+		Input any
+		Err   string
+	}{
+
+		{
+			Name:  "invalid input integer",
+			Input: 7,
+			Err:   "unexpected type for bytes: int",
+		},
+		{
+			Name:  "invalid input fixed array",
+			Input: [3]byte{0x01, 0x02, 0x03},
+			Err:   "unexpected type for bytes: [3]uint8",
+		},
+	} {
+		t.Run(tv.Name, func(t *testing.T) {
+			_, err := NewBytesInstance(tv.Input)
+			assert.EqualError(t, err, tv.Err)
+		})
+	}
+}
+
+func TestInstance_MarshalCBOR_Bytes(t *testing.T) {
+	tv, err := NewBytesInstance(TestBytes)
+	require.NoError(t, err)
+	// 560 (h'458999786556')
+	// tag(560): d9 0230
+	expected := MustHexDecode(t, "d90230458999786556")
+
+	actual, err := tv.MarshalCBOR()
+	fmt.Printf("CBOR: %x\n", actual)
+
+	assert.Nil(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+func TestInstance_UnmarshalCBOR_Bytes_OK(t *testing.T) {
+	tv := MustHexDecode(t, "d90230458999786556")
+
+	var actual Instance
+	err := actual.UnmarshalCBOR(tv)
+
+	assert.Nil(t, err)
+	assert.Equal(t, "bytes", actual.Type())
+	assert.Equal(t, TestBytes, actual.Bytes())
+}
+
+func TestInstance_MarshalJSONBytes_OK(t *testing.T) {
+	testBytes := []byte{0x01, 0x02, 0x03}
+	tb := TaggedBytes(testBytes)
+	tv := Instance{&tb}
+	jsonBytes, err := tv.MarshalJSON()
+	require.NoError(t, err)
+	assert.Equal(t, `{"type":"bytes","value":"AQID"}`, string(jsonBytes))
+
+}
+
+func TestInstance_UnmarshalJSON_Bytes_OK(t *testing.T) {
+	for _, tv := range []struct {
+		Name  string
+		Input string
+	}{
+		{
+			Name:  "valid input test 1",
+			Input: `{ "type": "bytes", "value": "MTIzNDU2Nzg5" }`,
+		},
+		{
+			Name:  "valid input test 2",
+			Input: `{ "type": "bytes", "value": "deadbeef"}`,
```

Comment:
this example is slightly confusing as it gives the impression that base-16 is a valid format :-)